nursepy/eda.py

```diff
@@ -39,17 +34,27 @@ def eda(input_data):
               max     1.867558}}
     """
 
+    if not isinstance(input_data, pd.core.frame.DataFrame):
+        raise ValueError('input_data must be instance of pandas.core.frame.DataFrame.')
+
+    if input_data.empty:
+        raise ValueError('input_data should contain at least one axis.')
+
+    if not all(type(col) == str for col in input_data.columns):
+        raise ValueError('All column names should be string.')
+
     eda_summary = defaultdict(dict)
 
     # for each column, calculate altair histogram and descriptive statistics
     for column_name in input_data.columns:
+        suffix = ":O" if str(input_data[column_name].dtype) in ['object', 'bool'] else ":Q"
         column_specific_hist = alt.Chart(input_data).mark_bar().encode(
```

Comment:
might it be useful to customize the altair plots?

Reply:
I think leaving customization to the library user makes more sense. I tried to keep it as simple as possible while handling numerical/categorical values. Library users can choose to change default chart settings like the number of bins or font sizes.